release/cmd/flags.go

```diff
@@ -14,43 +14,396 @@
+	"github.com/projectcalico/calico/release/internal/utils"
+	"github.com/projectcalico/calico/release/pkg/manager/operator"
+)
+
+// globalFlags are flags that are used across all commands
+var globalFlags = append([]cli.Flag{debugFlag}, append(ciFlags, slackFlags...)...)
+
+// debugFlag is a flag used to enable verbose log output
+var debugFlag = &cli.BoolFlag{
+	Name:        "debug",
+	Aliases:     []string{"d"},
+	Usage:       "Enable verbose log output",
+	Value:       false,
+	Destination: &debug,
 }
+
+// Product repository flags are flags used to interact with the product repository
+var (
+	gitFlags     = []cli.Flag{orgFlag, repoFlag, repoRemoteFlag}
+	productFlags = append(gitFlags, []cli.Flag{
+		releaseBranchPrefixFlag,
+		devTagSuffixFlag,
+	}...)
+
+	// Git flags for interacting with the git repository
+	orgFlag = &cli.StringFlag{
+		Name:    "org",
+		Usage:   "The GitHub organization to use for the release",
+		EnvVars: []string{"ORGANIZATION"},
+		Value:   utils.ProjectCalicoOrg,
+	}
+	repoFlag = &cli.StringFlag{
+		Name:    "repo",
+		Usage:   "The GitHub repository to use for the release",
+		EnvVars: []string{"GIT_REPO"},
+		Value:   utils.Calico,
+	}
+	repoRemoteFlag = &cli.StringFlag{
+		Name:    "remote",
+		Usage:   "The remote for the git repository",
+		EnvVars: []string{"GIT_REMOTE"},
+		Value:   utils.DefaultRemote,
+	}
+
+	// Branch/Tag flags are flags used for branch & tag management
+	mainBranchFlag = &cli.StringFlag{
+		Name:  "main-branch",
+		Usage: "The main branch to use for the release",
+		Value: utils.DefaultBranch,
+	}
+	releaseBranchPrefixFlag = &cli.StringFlag{
+		Name:    "release-branch-prefix",
+		Usage:   "The stardard prefix used to denote release branches",
+		EnvVars: []string{"RELEASE_BRANCH_PREFIX"},
+		Value:   "release",
+	}
+	devTagSuffixFlag = &cli.StringFlag{
+		Name:    "dev-tag-suffix",
+		Usage:   "The suffix used to denote development tags",
+		EnvVars: []string{"DEV_TAG_SUFFIX"},
+		Value:   "0.dev",
+	}
+	publishBranchFlag = &cli.BoolFlag{
+		Name:  "publish-branch",
+		Usage: "Push branch git. If false, all changes are local.",
+		Value: true,
+	}
+	newBranchFlag = &cli.StringFlag{
+		Name:  "branch-stream",
+		Usage: fmt.Sprintf("The new major and minor versions for the branch to create e.g. vX.Y to create a %s-vX.Y branch", releaseBranchPrefixFlag.Value),
+	}
+)
+
+// Validation flags are flags used to control validation
+var (
+	skipValidationFlag = &cli.BoolFlag{
+		Name:    "skip-validation",
+		Usage:   "Skip all validation while performing the action",
+		EnvVars: []string{"SKIP_VALIDATION"},
+		Value:   false,
+	}
+)
+
+// Container image flags are flags used to control container image building and publishing
+var (
+	registryFlag = &cli.StringSliceFlag{
+		Name:    "registry",
+		Usage:   "Override default registries for the release. Repeat for multiple registries.",
+		EnvVars: []string{"REGISTRIES"}, // DEV_REGISITRIES is already used by the build system.
```

Comment:
I do not think it is a good idea to use the same variable name used in `lib.Makefile`

Reply:
Could you explain why though? It seems to me that if one of our makefiles calls this tooling with an env var set, it would make sense for them to match?

I think the problem isn't whether or not our Makefile env vars are using the same values as our tool env vars. If the env vars have the same meaning, then they should be the same value. There's no point having a bunch of different env vars that all do the same thing (in this case, configure which registries to use) but have different names.

If env vars have the same name but have different meanings, that is the problem.